src/fhir.js

```
@@ -102,13 +106,80 @@ class PatientSource {
   }
 }
 
+class AsyncPatientSource {
+  constructor(filePathOrXML, serverUrl) {
+    this._index = 0;
+    this._patientIds = [];
+    this._modelInfo = load(filePathOrXML);
+    this.fhirServer = axios.create({
+      baseURL: serverUrl,
+      timeout: 10000,
+      headers: {
+        'Content-Type': 'application/fhir+json',
+        accept: 'application/fhir+json'
+      }
+    });
+  }
+
+  // Convenience factory method for getting a FHIR 1.0.2 (DSTU2) Patient Source
+  static FHIRv102(serverUrl) {
+    return new AsyncPatientSource(FHIRv102XML, serverUrl);
+  }
+
+  // Convenience factory method for getting a FHIR 3.0.0 (STU3) Patient Source
+  static FHIRv300(serverUrl) {
+    return new AsyncPatientSource(FHIRv300XML, serverUrl);
+  }
+
+  // Convenience factory method for getting a FHIR 4.0.0 (R4) Patient Source
+  static FHIRv400(serverUrl) {
+    return new AsyncPatientSource(FHIRv400XML, serverUrl);
+  }
+
+  // Convenience factory method for getting a FHIR 4.0.1 (R4) Patient Source
+  static FHIRv401(serverUrl) {
+    return new AsyncPatientSource(FHIRv401XML, serverUrl);
+  }
+
+  loadPatientIds(ids) {
+    this._patientIds = this._patientIds.concat(ids);
+  }
+
+  async currentPatient() {
+    if (this._index < this._patientIds.length) {
+      const id = this._patientIds[this._index];
+      const response = await this.fhirServer.get(`/Patient/${id}`);
+      if (response.status !== 200) {
+        throw new Error(
+          `Unable to retrieve Patient/${id} from server. Responded with error code: ${response.status}`
```

I'm wondering if there's a way we can get more error details here. I believe axios will throw an error when a non-2XX error code is returned. So maybe instead of this if/else, we use `try/catch` and catch the error? Then we can re-throw an error as you've done here, but include `e.message` instead of `response.status`? `e.message` will usually include the status code anyway